### ISSUE TITLE
Provide warning for using roles that aren't fully supported in browsers

### ISSUE THREAD

**Issue by backwardok** [w3c/aria-practices#1150]
There are some roles (like `slider` and `spinbutton`) that have related accessibility events (like `increment` and `decrement`) that are not exposed in browser APIs currently. Native equivalents (`<input type="range">` and `<input type="number">` respectively) have some support (when testing with macOS VoiceOver in Safari, the range accepts increment/decrement events, but number does not), but when using the ARIA pattern, this results in making the request to increment/decrement but no resulting change in behavior (since there's no JavaScript event to listen to in order to add the behavior in). 

It would be helpful to add warnings for some of the patterns that result in allowing AT events that cannot be supported currently in browsers.

**Comment by jnurthen** [comment]
I think this is a duplicate of https://github.com/w3c/aria-practices/issues/8

**Comment by backwardok** [comment]
It's kind of related to #8, but doesn't just affect touch devices. Even when using a keyboard + desktop SR, the same problem exists. 

In chatting about this with @scottaohara , he suggested adding extra information on the gaps of certain patterns on the example pages.

**Comment by jnurthen** [comment]
Yeah ok - it is more VO (and maybe Talkback too I can't remember whether they fire up/down key events on increment/decrement off-hand)

**Comment by scottaohara** [comment]
to clarify, i didn't mean to indicate that i think all gaps should be accounted for.  that'd be a lot of work/upkeep.

As i mentioned in #8 i'd be perfectly happy with a clear link/reference to sections 2.2 and/or 2.3 to remind developers "Testing assistive technology interoperability is essential before using code from this guide in production"

**Comment by a11ydoer** [comment]
https://www.w3.org/2019/08/06-aria-apg-minutes.html#item03

**Comment by mcking65** [comment]
We discussed this in the August 6, 2019 APG task Force meeting. See [Minutes for topic APG language for new ARIA features](https://www.w3.org/2019/08/06-aria-apg-minutes.html).

Note that browser and assistive technology support are outside the scope of the APG. Browser support is addressed by automated testing during the ARIA candidate recommendation process and assistive technology support is being addressed by the w3c/aria-at project.

Sometime next year, we may have the capability to link to assistive technology support data from the w3c/aria-at project from the APG.

In the mean time, as Scott points out, we refer people to the need for thorough testing as described in section 2.3.

**Comment by mcking65** [comment]
This was discussed again on [September 10, 2019](https://www.w3.org/2019/09/10-aria-apg-minutes.html).

We will consider a PR that scopes a warning statement to alerting readers to support gaps that can only be resolved by development of a new technology, such as AOM. These gaps are distinctly different from browser or assistive technology implementation bugs, which will be covered by the aria-at project. The aria-at project objective is to provide assistive technology support tables that can be integrated into each example page.

In the September 10 meeting, @charmarkk suggested he might be able to draft a PR for the 1.2 release.

**Comment by mcking65** [comment]
Commit eec5af6 resolves this for slider patterns.

We now need to decide if we need it on any other patterns. There are some useful comments in PR #1186. This is a topic for the [February 4 meeting](https://github.com/w3c/aria-practices/wiki/February-4%2C-2020-Meeting).

**Comment by carmacleod** [comment]
[Discussion from Feb 4 meeting](https://www.w3.org/2020/02/04-aria-apg-minutes.html#item01).

**Comment by mcking65** [comment]
The warning about API gaps has been added to slider and spin buttons. All examples now also have stronger language related to testing mobile support.

I think we can close this now.
